src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,45 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px ;
+  padding: 25px;
+  margin: auto;
+  width: (7 * $cell-width) + 10px;
+
+  @each $day, $number in $days {
+    &--start-day-#{$day} :first-child {
+      margin-left: ($cell-width + 1px) * $number;
+    }
+  }
+
+  @for $value from 28 through 31 {
+    &-month-length--#{$value} :nth-child(n + #{$value + 1}) {
+      display: none;
+    }
+  }
+}
+
+.calendar__day {
+  width: $cell-width;
+  height: $cell-height;
+  display: block;
+  box-sizing: border-box;
+  background-color: #eee;
+  border: 1px solid black;
+  font-size: 30px;
+  text-align: center;
+  transition: 0.5s;
+
+  &:hover {
+    background-color: pink;
+    transform: translateY(-20px);
+    cursor: pointer;
+  }
+
+  @for $day from 1 through 31 {
+    &--day#{$day}::before {
+      content: "#{$day}";
+      line-height: $cell-height;
+    }
```

Comment:
you could simplify it quite a bit
```
    @for $day from 1 through 31 {
      &:nth-child(#{$day})::before {
        content: "#{$day}";
				...
      }
    }
```

and then you won't need the whole class `calendar__day--day1...31`

let's make it work @Grayfox147

Reply:
Ahhh I see! if I use the nth-child $day I can replace the code from HTML